dwa_local_planner/src/dwa_planner_ros.cpp

```diff
@@ -184,6 +184,24 @@ namespace dwa_local_planner {
       return false;
     }
 
+    // change limits if dist_tolerance or angle_tolerance are set, and different from the current values
+    if ((dist_tolerance > 0 && dist_tolerance != planner_util_.getCurrentLimits().xy_goal_tolerance) ||
+        (angle_tolerance > 0 && angle_tolerance != planner_util_.getCurrentLimits().yaw_goal_tolerance)) {
+      std::lock_guard<std::mutex> lock(config_mtx_);
+      auto limits = planner_util_.getCurrentLimits();
+      if (dist_tolerance > 0) {
+        ROS_INFO_STREAM_COND(dist_tolerance != limits.xy_goal_tolerance,
```

Comment:
S: Won't this be super spammy? Might be better to change to DEBUG. Same thing for the other log below

Reply:
This will only get printed if the global planner uses up part of the total goal tolerance (which means the goal is in an obstacle), so this shouldn't happen too often. I can still change it to debug though

Reply:
one other thing to note is that these goal tolerance changes aren't reflected in the parameter values of the dynamically reconfigured parameters (i.e. these changes wouldn't be seen when looking at rqt); we could do that by calling an actual dynamic reconfigure call to itself, but that seems pretty ugly; so I thought a print is maybe a decent trade-off

Reply:
> This will only get printed if the global planner uses up part of the total goal tolerance

What do you mean? Won't this get printed every time mbf calls `isGoalReached` and the tolerances are different? What does global planner have to do with this?

> we could do that by calling an actual dynamic reconfigure call to itself,

I was writing an answer to this when I noticed a problem with this PR.

What we want to achieve is have 2 possible options:
1. tolerance from mbf (which can be from action or not, planner doesn't
3. tolerance from planner (reconf)

With the current implementation you're setting the tolerance from reconfigure to the one given from mbf (even if it's not actually reflected in rqt).

Isn't this essentially getting rid of option 2 and always using option 1? (assuming there's no reconfigure)

Reply:
Also, this is a pretty big change in behavior -- most of the time we're not longer using the tolerance from the planner param but from the mbf param. Should maybe write this somewhere

Reply:
I improved the documentation of `isGoalReached` in the commit [improve documentation](https://github.com/rapyuta-robotics/ros-navigation/pull/71/commits/555287f170f1523284dc35cca4dbbfe7b8ba3e96)